apps-rendering/src/server/footballContent.ts

```diff
@@ -90,44 +186,32 @@ const teamsFromTags = (tags: Tag[]): Option<Teams> => {
 
 const getFootballContent = async (
 	content: Content,
-): Promise<FootballContent | undefined> => {
+): Promise<Result<string, FootballContent>> => {
 	const teams = teamsFromTags(content.tags);
 
-	if (teams.kind === OptionKind.Some) {
-		const currentTeams = teams.value;
-		const teamA = some(currentTeams[0]);
-		const teamB = some(currentTeams[1]);
-
-		const webPublicationDate = content.webPublicationDate?.iso8601;
+	const makeDate = (s: string): Date => new Date(s);
 
-		if (webPublicationDate) {
-			const date = some(new Date(webPublicationDate));
+	const date = pipe(
+		content.webPublicationDate?.iso8601,
+		fromNullable,
+		optMap(makeDate),
+	);
 
-			const selectorId = getFootballSelector(date, teamA, teamB);
+	const selectorId = map2(getFootballSelector)(date)(teams);
 
-			if (selectorId.kind === OptionKind.Some) {
-				const footballEndpoint = getFootballEndpoint(selectorId);
+	if (selectorId.kind === OptionKind.Some) {
```

Comment:
Seems a bit strange to be comparing to `OptionKind.Some` outside of `lib.ts`.  Can you not fold over an option?

Reply:
Struggling a bit with this one, I haven't really used `fold` before but if i do the following is that correct?
```
	return fold(async (selectorIdValue: string) => {
		const footballEndpoint = getFootballEndpoint(selectorIdValue);

		const response = await fetch(footballEndpoint);

		const footballContent = await parseFootballResponse(
			response,
			selectorIdValue,
		);

		return footballContent;
	}, err('Could not get selectorId'))(selectorId);
	```

Reply:
I'm getting an error for the return type but i'm not sure why this is different now.
```
Argument of type 'Err<string>' is not assignable to parameter of type 'Promise<Result<string, FootballContent>>'.

Reply:
The second argument of `fold` has to be of the same type as the return type of the first argument, so in this case `Promise<Result<string, FootballContent>>`.  

I think you should be able to change  `err('Could not get selectorId')` to `Promise.resolve(err('Could not get selectorId'))`

Reply:
Great, thanks. That's updated now.